scalapb-runtime/jvm/src/main/scala/scalapb/StructUtils.scala

```diff
@@ -0,0 +1,99 @@
+    case Kind.NumberValue(v) if fd.scalaType == ScalaType.Int && v.isValidInt => Right(PInt(v.toInt))
+    case Kind.StringValue(v) if fd.scalaType == ScalaType.Long =>
+      Try {
+        PLong(v.toLong)
+      } match {
+        case Success(pLong) => Right(pLong)
+        case Failure(_) => Left(StructDeserError(fd.number, s"Invalid value for long: '$v'"))
+      }
+    case Kind.NumberValue(v) if fd.scalaType == ScalaType.Double => Right(PDouble(v))
+    case Kind.NumberValue(v) if fd.scalaType == ScalaType.Float => Right(PFloat(v.toFloat))
+    case Kind.StringValue(v) if fd.scalaType == ScalaType.ByteString => Right(PByteString(ByteString.copyFrom(v, Charset.forName("UTF-8"))))
+    case Kind.StringValue(v) if fd.scalaType.isInstanceOf[ScalaType.Enum] =>
+      val enumDesc = fd.scalaType.asInstanceOf[ScalaType.Enum].descriptor
+      enumDesc.values
+        .find(_.name == v)
+        .map(PEnum)
+        .toRight(StructDeserError(fd.number, s"""Expected Enum type "${enumDesc.fullName}" has no value named "$v""""))
+    case Kind.StringValue(v) if (fd.scalaType == ScalaType.String) => Right(PString(v))
+    case Kind.BoolValue(v) if (fd.scalaType == ScalaType.Boolean) => Right(PBoolean(v))
+    case Kind.ListValue(v) if (fd.isRepeated.asInstanceOf[Boolean]) =>
+      flatten(v.values.map(fromValue(fd, companion)))
+        .right.map(PRepeated)
+    case Kind.StructValue(v) if (fd.scalaType.isInstanceOf[ScalaType.Message]) =>
+      structMapToFDMap(companion.messageCompanionForFieldNumber(fd.number), v.fields).right.map(PMessage)
+    case Kind.Empty => Right(PEmpty)
+    case kind: Kind => Left(StructDeserError(fd.number, s"Field `${fd.fullName}` is of type '${fd.scalaType}' but received '$kind'"))
+  }
+
+  private def defaultFor(fd: FieldDescriptor): PValue =
+    if (fd.isRepeated.asInstanceOf[Boolean])   //TODO @thesmaet- do we have a better alternative?
+      PRepeated(Vector.empty)
+    else
+      PEmpty
+
+  private def toStruct(pMessageMap: Map[FieldDescriptor, PValue]): Struct = {
+    val pMessageWithNonEmptyFields = pMessageMap.filter { case (_, value) =>
+      value != PEmpty && value != PRepeated(Vector.empty)
+    }
+    Struct(pMessageWithNonEmptyFields.map(pair => pair._1.name -> toValue(pair._2)))
+  }
+
+  private def toValue(pValue: PValue): Value = Value(pValue match {
+    case PInt(value) => Value.Kind.NumberValue(value.toDouble)
+    case PLong(value) => Value.Kind.StringValue(value.toString)
+    case PDouble(value) => Value.Kind.NumberValue(value)
+    case PFloat(value) => Value.Kind.NumberValue(value.toDouble)
+    case PString(value) => Value.Kind.StringValue(value)
+    case PByteString(value) => Value.Kind.StringValue(value.toStringUtf8) //TODO(@thesamet)- not sure if we have a better option here. WDYT?
+    case PBoolean(value) => Value.Kind.BoolValue(value)
+    case PEnum(value) => Value.Kind.StringValue(value.name.toString) //TODO(@thesamet)- value.name returns Any. Is toString ok?
```

Comment:
EnumValueDescription#name is a string - can you check why you're seeing Any?

Reply:
I think it's sbt/intellij issues. It shows red in IJ but builds fine 🤷